source/common/http/path_utility.cc

```diff
@@ -54,35 +95,38 @@ bool PathUtil::canonicalPath(RequestHeaderMap& headers) {
   if (!query_suffix.empty()) {
     normalized_path.insert(normalized_path.end(), query_suffix.begin(), query_suffix.end());
   }
-  headers.setPath(normalized_path);
-  return true;
+  return normalized_path;
 }
 
-void PathUtil::mergeSlashes(RequestHeaderMap& headers) {
-  ASSERT(headers.Path());
-  const auto original_path = headers.getPathValue();
-  // Only operate on path component in URL.
-  const absl::string_view::size_type query_start = original_path.find('?');
-  const absl::string_view path = original_path.substr(0, query_start);
-  const absl::string_view query = absl::ClippedSubstr(original_path, query_start);
-  if (path.find("//") == absl::string_view::npos) {
-    return;
+PathTransformer::PathTransformer(envoy::type::http::v3::PathTransformation path_transformation) {
+  const auto& operations = path_transformation.operations();
+  std::vector<uint64_t> operation_hashes;
+  for (auto const& operation : operations) {
+    uint64_t operation_hash = MessageUtil::hash(operation);
+    if (find(operation_hashes.begin(), operation_hashes.end(), operation_hash) !=
+        operation_hashes.end()) {
+      throw EnvoyException("Duplicate path transformation");
```

Comment:
nit: add comment that we don't expect duplicate transforms for merge slashes and RFC path normalization (so that if this is extended it can be revisited)

Reply:
Thanks, will add comment!